GGRC-4258 Use own instance in modals controller

romamatusevich opened the pull request:
# Issue description

This issue is actual for any modal in edit mode. In case of having modal in edit mode and changing a field's value it's possible to see that the value for the same field at the original page is changed too.

# Steps to test the changes

1. Create an instance (control, assessment, etc.)
2. Invoke edit modal (control, assessment, etc.) window
3. While typing a field's value (for example: description) look at the same field at the page: the same value is shown

# Solution description

## Technical description of the issue
Modals controller contains `instance` field. Looks like initially it was designed to create and handle own instance. Actually modals controller uses `instance` from `options` directly and methods like `set_value`, `set_value_from_element`, etc. change it instead of using own `instance` of the controller. This is the reason why during typing a new value for any field it will be applied not only for the modal in edit mode but also for the page from which this modal was invoked. Also this makes modals controller tightly coupled with `options.instance`.

## Solution
The idea is to use own `instance` of the modals controller instead of changing `instance` from `options` directly. This allows to decouple modals controller and `options.instance`. During initialization phase in `fetch_data` own instance is created and shallow fields of `options.instance` are copied. On save the fields from own `instance` are copied to `options.instance` back.

## Another attempts to solve this issue
- Create own instance in `init` method only. In such a case `options.instance` will be passed in `fetch_templates` method and there will be no synchronization between own `instance` and `options.instance`.
- Save own `instance` in `save_instance` method instead of copying the fields from own instance to `options.instance`. It works for a lot of pages like control, program, etc. but it does not work some pages like assessment. After investigation I found that assessment contains event listeners and they are added to `options.instance`.

# Sanity checklist

- [x] I have clicked through the app to make sure my changes work and not break the app.
- [ ] I have applied the correct milestone and labels.
- [x] My changes fix the issue described in the description (and do nothing else). 🤞
- [x] My changes are covered by tests.
- [x] My changes follow our [performance guidelines](https://github.com/google/ggrc-core/blob/dev/docs/source/contributing/performance.rst).
- [x] My changes follow our [js](https://github.com/google/ggrc-core/blob/dev/docs/source/contributing/javascript.rst) and/or [python](https://github.com/google/ggrc-core/blob/dev/docs/source/contributing/python.rst) guidelines.
- [x] My commits follow our [commit guidelines](https://github.com/google/ggrc-core/blob/dev/docs/source/contributing/git/how_to_write_a_commit_message.rst).

# PR Review checklist

- [ ] The changes fix the issue and don't cause any apparent regressions.
- [ ] Labels and milestone are correctly set.
- [ ] The solution description matches the changes in the code.
- [ ] There is no apparent way to improve the performance & design of the new code.
- [ ] The pull request is opened against the correct base branch.
- [ ] Upon merging, the Jira ticket's fixversion is correctly set and the ticket is moved to "QA - In Progress".

# TODO

- [ ] Locally some automation (ui) tests were failed and I am going to investigate the reason

YuriChupyrkin commented:
@romamatusevich , do we need to restore instance after **discard** modal?

Please take a look at **modal-ajax.js**. This file contains follow source code:
`$target.on('modal:discard', () => {
        if ( instance ) {
          instance.restore(true);
        }
      });
`

Seems like this piece of code is unneeded now. Please check this

romamatusevich commented:
@YuriChupyrkin,

Right, we use own instance in modal controller and don't need to restore options model. Thank you for the suggestion. I removed this listener and updated the pull request. Eslint, unit tests, automation tests locally were fine.
